c2rust-analyze/src/rewrite/expr/mir_op.rs

```diff
@@ -266,25 +276,32 @@ impl<'a, 'tcx> ExprRewriteVisitor<'a, 'tcx> {
                         self.visit_slice_as_ptr(elem_ty, &args[0], pl_ty);
                         return;
                     }
-                    _ => {}
-                }
 
-                // General case: cast `args` to match the signature of `func`.
-                let poly_sig = func_ty.fn_sig(tcx);
-                let sig = tcx.erase_late_bound_regions(poly_sig);
-
-                for (i, _op) in args.iter().enumerate() {
-                    if i >= sig.inputs().len() {
-                        // This is a call to a variadic function, and we've gone past the end of
-                        // the declared arguments.
-                        // TODO: insert a cast to turn `op` back into its original declared type
-                        // (i.e. upcast the chosen reference type back to a raw pointer)
-                        continue;
+                    Callee::LocalDef { def_id, substs: _ } => {
+                        // TODO: handle substs (if nonempty)
+                        if let Some(lsig) = self.acx.gacx.fn_sigs.get(&def_id) {
+                            self.enter_rvalue(|v| {
+                                for (i, op) in args.iter().enumerate() {
+                                    if let Some(&lty) = lsig.inputs.get(i) {
+                                        v.enter_call_arg(i, |v| v.visit_operand(op, Some(lty)));
+                                    } else {
+                                        // This is a call to a variadic function, and we've gone
+                                        // past the end of the declared arguments.
+                                        // TODO: insert a cast to turn `op` back into its original
+                                        // declared type (i.e. upcast the chosen reference type
+                                        // back to a raw pointer)
+                                        continue;
+                                    }
+                                }
```

Comment:
```suggestion
                                let (args, _variadic_args) = args.split_at(lsig.inputs.len());
                                for (i, (op, &lty)) in iter::zip(args, lsig.inputs).enumerate() {
                                    v.enter_call_arg(i, |v| v.visit_operand(op, Some(lty)));
                                }
                                // TODO: insert a cast to turn `op` back into its original
                                // declared type (i.e. upcast the chosen reference type
                                // back to a raw pointer)
```

I think this would be clearer (and simpler) about the variadic args being the extra args at the end rather than matching on `slice::get`.